pkg/daemon/ceph/osd/daemon.go

```diff
@@ -84,10 +86,32 @@ func StartOSD(context *clusterd.Context, osdType, osdID, osdUUID, lvPath string,
 		if err := releaseLVMDevice(context, volumeGroupName); err != nil {
 			return fmt.Errorf("failed to release device from lvm. %+v", err)
 		}
+
 	}
+
 	return nil
 }
 
+func handleTerminate(context *clusterd.Context, volumeGroupName string) error {
+	sigc := make(chan os.Signal, 1)
+	signal.Notify(sigc, syscall.SIGTERM)
+	for {
+		select {
+		case <-sigc:
+			logger.Infof("shutdown signal received, exiting...")
+			// shut down the osd-ceph process so that lvm release does not show device in use error.
+			if err := context.Executor.ExecuteCommand(false, "", "killall", "ceph-osd"); err != nil {
```

Comment:
@travisn isn is there better way to manually delete the ceph-osd process.  The use case here is that when the use terminates the OSD pods, the process is still attached to the device file and as a result `lvchange -an` fails to release the device. `killall ceph-osd` won't work when there are multiple osds on the same name. So wanted to check if there is straight forward way to do it.

Reply:
If we're capturing the SIGTERM, the ceph-osd process would also be responding to that and shutting down automatically, right? Can we just wait here for the process to exit? Each osd has arguments (such as osd id) that we could detect the appropriate ceph-osd process to watch. But I don't see why we would need to kill the osd process here.

Reply:
Thing is if you do `killall ceph-osd` that going to kill **ALL** the OSDs on that machine since we use `pid=host`. So we must isolate the OSD to kill.
They are probably a million different ways to do this but here is one:

* use the `ROOK_OSD_ID` to find the OSD to kill
* correlate the id with any process running `ceph-osd --foreground --id $ROOK_OSD_ID` and get its PID.

Hope this helps.